### PR TITLE
Remove collections dependency and es6-set instead

### PR DESCRIPTION
**Pull request by faridnsh**
Since [collections module](http://collectionsjs.com) modifies native objects, it's only good
to be used as a top dependency of an application and not modules that
it's using.

The problem is that this will conflict if native objects has been
modified by another module, let's say something like [sugar](http://sugarjs.com) or just
another version of collections. When you have two of these being used in
your app, you might be in trouble, where some code depends on a
behaviour of a method implemented by one module but actually the method
had been silently overriden by another module having a different
behaviour.

### CODE REVIEW

File: http-apps/negotiate.js [hunk 1]
```diff
@@ -23,8 +23,8 @@ exports.Method = function (methods, methodNotAllowed) {
         methodNotAllowed = Status.methodNotAllowed;
     return function (request) {
         var method = request.method;
-        if (Object.has(keys, method)) {
-            return Object.get(methods, method)(request);
+        if (keys.indexOf(method) !== -1) {
+            return methods[method](request);
```
Comment (kriskowal) [inline http-apps/negotiate.js:26]: O(n) vs O(1). Let’s use Object.prototype.hasOwnProperty.call.

**Comment by kriskowal** [comment]
With version 2 of Collections, I’m moving away from shims in favor of polymorphic operators (e.g., pop-swap, pop-equals, pop-compare). Q-IO version 2 already depends on Collections, although the change is incomplete. It makes me sad for Q-IO v1 to diverge, but this does seem like a sensible change.

File: http-apps/negotiate.js (hunk 1 above)
Comment (faridnsh) [inline http-apps/negotiate.js:26]: And one less line... Fixed it :)

**Comment by hthetiot** [comment]
+1 merge pls

**Comment by kriskowal** [comment]
Adding @hthetiot as a collaborator. Send me your npm handle and I'll grant you publish as well.

**Comment by hthetiot** [comment]
@alFReD-NSH Can you rebase ?
simply ignore my collections update in package.json.
Also if we merge that should we name that v3 ? @kriskowal

**Comment by hthetiot** [comment]
Thanks @alFReD-NSH for great work.
On v1 (master) i have couple of fixes that i merged into a new PR here
- https://github.com/kriskowal/q-io/pull/168

Feel free to re-open this PR with more changes if needed, in the meantime we may merge #168

**Pull request closed without merging**